app/main.py

```diff
@@ -1,6 +1,26 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached = []
+    arg_cache = []
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        nonlocal cached, arg_cache
```

Comment:
this statement is redundant, as dict is mutable type

Reply:
I do not use dict

Reply:
Lists are also mutable, so mind this comment